### DISCUSSION TITLE
API for Dahua TiOC?

### DISCUSSION THREAD

**Discussion by Cold-Lemonade** [blakeblackshear/frigate#6080] [General]
My question is not directly related to Frigate, but I suspect there may be some Frigate users who know how to accomplish what I'm trying to do.

I have two Dahua TiOC cameras (actually they are clones... IPC-T5241H-AS-PV). These cameras are great because my Home Assistant can tell them turn on their alarms if Frigate detects a person. This is quite useful for me because there are a lot of teenagers on my block that like to hangout in front of my house at night--and the alarm chases them away.

These cameras have several pre-installed audio files that can be used for the alarm. Right now, I just have it play the siren wav file. But there are others, like one saying "You are trespassing. Leave now." I would like Home Assistant to tell the camera to switch the audio file it plays when the siren is engaged. To do this, I need to know the Dahua API command to issue. Does anyone know what it is?

Alternatively, is there another way to accomplish what I'm trying to do?

**Comment by NickM-27** [comment]
Your camera supports two way talk per the other conversation. You can use the go2rtc api or WebRTC card integration to play any audio file on the camera.

**Reply by Cold-Lemonade** [reply]
Many thanks @NickM-27 You've been incredibly kind in helping. Where should I go to read more about these two methods to plays any audio file on my cameras?

**Reply by NickM-27** [reply]
Go2rtc docs

**Comment by yeahman45** [comment]
> My question is not directly related to Frigate, but I suspect there may be some Frigate users who know how to accomplish what I'm trying to do.
> 
> I have two Dahua TiOC cameras (actually they are clones... IPC-T5241H-AS-PV). These cameras are great because my Home Assistant can tell them turn on their alarms if Frigate detects a person. This is quite useful for me because there are a lot of teenagers on my block that like to hangout in front of my house at night--and the alarm chases them away.
> 
> These cameras have several pre-installed audio files that can be used for the alarm. Right now, I just have it play the siren wav file. But there are others, like one saying "You are trespassing. Leave now." I would like Home Assistant to tell the camera to switch the audio file it plays when the siren is engaged. To do this, I need to know the Dahua API command to issue. Does anyone know what it is?
> 
> Alternatively, is there another way to accomplish what I'm trying to do?

how did to manage to trigger the alarm on the TIOC cameras?

**Reply by Cold-Lemonade** [reply]
I access the Dahua API by issuing a curl command from the cli in Home Assistant. Here's the basic code that you'll need to adjust with your own username, password, and ip address:

```
curl --digest -g --interface enp0s3 "http://username:password@ipaddress/cgi-bin/coaxialControlIO.cgi?action=control&channel=1&info[0].Type=2&info[0].IO=1"
```

There is more about the Dahua API over at ipcamtalk.com.

Good luck.

**Reply by yeahman45** [reply]
> I access the Dahua API by issuing a curl command from the cli in Home Assistant. Here's the basic code that you'll need to adjust with your own username, password, and ip address:
> 
> ```
> curl --digest -g --interface enp0s3 "http://username:password@ipaddress/cgi-bin/coaxialControlIO.cgi?action=control&channel=1&info[0].Type=2&info[0].IO=1"
> ```
> 
> There is more about the Dahua API over at ipcamtalk.com.
> 
> Good luck.

thx! I don't find any command to disarm the alarm through the API; do you by any chance have found it?

**Comment by Cold-Lemonade** [comment]
Try changing the last zero to one.

**Comment by Cold-Lemonade** [comment]
I mean the last one to zero.
